src/plugins/intel_gna/src/backend/gna_limitations.hpp

```diff
@@ -309,14 +308,14 @@ class Limitations {
     bool m_use_only_16bit_conv_weights = false;
     size_t m_mem_alignment = 0;
     std::shared_ptr<cnn2d::AbstractValidator> m_cnn_validator;
-    static thread_local std::shared_ptr<Limitations> k_instance;
+    static std::unique_ptr<Limitations> kInstance;
 };
 
-inline std::shared_ptr<Limitations> Limitations::get_instance() {
-    if (!k_instance) {
+inline Limitations& Limitations::get_instance() {
```

Comment:
is it cherry-pick of some tested solution? Looks like `singleton` pattern may be used here to prevent some leaks

Reply:
The final solution is implemented in https://github.com/openvinotoolkit/openvino/pull/21273.
Please comment on it if you have any questions.